src/main/java/com/revature/services/PersonLogicImpl.java

```diff
@@ -87,4 +87,63 @@ public Person getPersonById(int id)  {
 		return dao.findAll();
 	}
 
+	@Override
+	public void savePerson(Person p) {
+		dao.save(p);
+	}
+
+	@Override
+	public Person updatePerson(Person p, String firstname, String lastname, int role) {	
+		
+		if (!firstname.equals("")) {
+			
+			 System.out.println("changing first to " + firstname);
+			 p.setFirstName(firstname);
+			 
+		} else {
+			
+			System.out.println("no change to first " + p.getFirstName());
+			p.setFirstName(p.getFirstName());
+		}
+		
+		
+		
+		if (!lastname.equals("")) {
+			
+			 System.out.println("changing last to " + lastname);
+			 p.setLastName(lastname);
+			 
+		} else {
+			
+			System.out.println("no change to last " + p.getLastName());			
+			p.setLastName(p.getLastName());
+		}
+		
+		
+	   
+		if (role != 0) {
+			
+			System.out.println("changing role to " + role);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AS106)